models/User.go

```diff
@@ -729,6 +732,26 @@ func SubtractUserCash(user *User, fee uint64, tx *gorm.DB) error {
 	return nil
 }
 
+// AddUserReservedCash subtracts reservedCash for users account
+// This function should be called ONLY AFTER lock is obtained on user
+func AddUserReservedCash(user *User, fee uint64, tx *gorm.DB) error {
```

Comment:
Why are you calling this parameter `fee`? It's not a fee at all. It's cash that's being reserved. Give it a more accurate name.